worker/src/lib/clients/S3Client.ts

```diff
@@ -70,4 +140,97 @@ export class S3Client {
       return { data: null, error: error?.message };
     }
   }
+
+  async uploadBase64ToS3(
+    base64Data: string,
+    assetType: string,
+    requestId: string,
+    orgId: string
+  ): Promise<string> {
+    const buffer = Buffer.from(base64Data, "base64");
+    const fileExtension = this.getFileExtension(assetType);
+    const uploadUrl = this.getRequestResponseImageUrl(
+      requestId,
+      orgId,
+      fileExtension
+    );
+
+    await this.uploadToS3(uploadUrl, buffer, assetType);
+    return uploadUrl;
+  }
+
+  async uploadImageToS3(
+    imageUrl: string,
+    requestId: string,
+    orgId: string
+  ): Promise<string> {
+    const response = await fetch(imageUrl);
+    if (!response.ok) {
+      console.error(`Failed to download image: ${response.statusText}`);
+      return "";
+    }
+    const blob = await response.blob();
+    const fileExtension =
+      this.getFileExtensionFromBlob(blob) ||
+      this.getFileExtensionFromUrl(imageUrl);
+    const uploadUrl = this.getRequestResponseImageUrl(
+      requestId,
+      orgId,
+      fileExtension ?? "jpg"
+    );
+
+    await this.uploadToS3(uploadUrl, await blob.arrayBuffer(), blob.type);
+    return uploadUrl;
+  }
+
+  private getFileExtensionFromBlob(blob: Blob) {
+    const mimeType = blob.type;
+    return mimeType.split("/").pop();
+  }
+
+  private getFileExtensionFromUrl(url: string) {
+    const pathname = new URL(url).pathname;
+    const lastSegment = pathname.split("/").pop();
+    return lastSegment?.split(".").pop();
+  }
+
+  private getFileExtension(mimeType: string): string {
+    const mimeExtensionMap = new Map<string, string>([
+      ["image/jpeg", "jpg"],
+      ["image/png", "png"],
+      ["image/gif", "gif"],
+      ["image/webp", "webp"],
+    ]);
+
+    return mimeExtensionMap.get(mimeType) || "jpg";
+  }
+
+  private extractBase64Data(dataUri: string): [string, string] {
+    const matches = dataUri.match(
+      /^data:(image\/(?:png|jpeg|jpg|gif|webp));base64,(.*)$/
+    );
+    if (!matches || matches.length !== 3) {
+      console.log("Invalid base64 image data");
+      return ["", ""];
+    }
+    return [matches[1], matches[2]];
+  }
+
+  private async uploadToS3(
```

Comment:
For example this method is perfect for living in the S3Client class.

Reply:
I agree here - should we move this?